sycl/include/sycl/ext/oneapi/device_global/device_global.hpp

```diff
@@ -42,9 +42,28 @@ struct HasArrowOperator<
 template <typename T, typename PropertyListT, typename = void>
 class device_global_base {
 protected:
-  T *usmptr;
-  T *get_ptr() noexcept { return usmptr; }
-  const T *get_ptr() const noexcept { return usmptr; }
+  using pointer_t = typename multi_ptr<T, access::address_space::global_space,
+                                       access::decorated::yes>::pointer;
+  pointer_t usmptr;
+  pointer_t get_ptr() noexcept { return usmptr; }
+  const pointer_t get_ptr() const noexcept { return usmptr; }
+
+public:
+  template <access::decorated IsDecorated>
+  multi_ptr<T, access::address_space::global_space, IsDecorated>
+  get_multi_ptr() noexcept {
+    __SYCL_HOST_NOT_SUPPORTED("get_multi_ptr()")
```

Comment:
We claim this method to be `noexcept`, but `__SYCL_HOST_NOT_SUPPORTED` is defined to throw an exception - it doesn't seem to be correct. Or am I missing something?

Reply:
We need it to compile but not run on host. As such I do not believe it is worth removing `noexcept` just in case someone using it invalidly.

Reply:
I do not suggest removing `noexcept`, especially if that is a part of our interface, but perhaps we should honor it since we defined it? We could just return `nullptr` on host and that would pretty much fine as "undefined behavior" and it would satisfy `noexcept`.

Reply:
That is definitely an option, but I wonder which would be preferred by a user. Returning a null multi_ptr may not be easy to track back, but a panic in this call should at least be easily findable, even if it is considered unrecoverable.